modules/admin/src/views/ListDocuments2.vue

```diff
@@ -126,18 +131,36 @@ export default {
       this.data = info.data
       this.listedProperties = columns || Object.keys(this.schema.properties)
     },
-    downloadCSV() {
+    downloadCSV(data = this.data) {
       const collection = this.collectionName
       let text = this.listedProperties.map((name) => '"' + name + '"').join(',') + '\n'
-      if (this.data) {
-        text += this.data.map((row) =>
+      if (data) {
+        text += data.map((row) =>
           this.listedProperties.map((key) => '"' + String(objectPath.get(row, key) || '').replace(/"/g, '""') + '"')
         ).join('\n')
         console.log(text)
         const blob = new Blob([text], { type: 'text/plain;charset=utf-8' })
         saveAs(blob, collection + '.csv')
       }
     },
+    async downloadAllCSV() {
+      let rows = []
+      let page = 1
+
+      while (this.count > rows.length) {
+        const params = {
+          page: page,
+          limit: 500,
+          orderby: '{"meta.created":-1}',
+        }
+
+        const response = (await request({ url: `/${this.collectionName}/`, params })).data
```

Comment:
please include the search query in this request

Reply:
@kane-mason  But this request is supposed to fetch all the data for **Download All** button. Wouldn't it be in-accurate to filter data in this case?

Reply:
the way i see it, is downloading all the pages of the current view - so its somewhat subjective, i need it to be like this so this is the way we go for now :)

Reply:
also this.count is calculated based on the filtered results, so not doing it this way will probably truncate the data when a filter is present

Reply:
Ah, I see. So I will include the search filters as well and everything should be fine.